src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -50,27 +225,296 @@ def execute(self, reqStatus):
+            msg = "%s: Workflows cleaned by pipeline: %s: %d"
+            self.logger.info(msg, self.currThreadIdent, pline.name, self.wfCounters['cleaned'][pline.name])
+        archived_num_requests = self.wfCounters['archived']
+        msg = "%s: Workflows archived: %d"
+        self.logger.info(msg, self.currThreadIdent, self.wfCounters['archived'])
+        return clean_num_requests, total_num_requests, archived_num_requests
+
+    def _dispatchWflow(self, wflow, *args, **kwargs):
+        """
+        A function intended to dispatch a workflow (e.g based on its status)
+        through one or more functional pipelines in case there is some more
+        complicated logic involved in the order we execute them but not just
+        a sequentially
+        """
+        # NOTE: The following dispatch logic is a subject to be changed at any time
+
+        if wflow['RequestType'] == 'Resubmission':
+            pass
+
+        # Clean:
+        elif wflow['RequestStatus'] in ['rejected', 'aborted-completed']:
+            # NOTE: We do not check the ParentageResolved flag for these
+            #       workflows, but we do need to clean output data placement
+            #       rules from the agents for them
+            for pline in self.agentlines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    break
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+        elif wflow['RequestStatus'] == 'announced' and wflow['ParentageResolved']:
+            for pline in self.cleanuplines:
+                try:
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline. Error: \n%s. "
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg, pline.name, str(ex))
+                    break
+                if wflow['CleanupStatus'][pline.name]:
+                    self.wfCounters['cleaned'][pline.name] += 1
+
+        # Archive:
+        try:
+            self.plineArchive.run(wflow)
+            self.wfCounters['archived'] += 1
+        except MSRuleCleanerArchival as ex:
+            msg = "%s: Archival Error: %s. "
+            msg += "\nWill retry again in the next cycle."
+            self.logger.error(msg, wflow['PlineMarkers'][-1], ex.myMessage)
+        except Exception as ex:
+            msg = "%s General error from pipeline. Error: \n%s. "
+            msg += "\nWill retry again in the next cycle."
+            self.logger.exception(msg, wflow['PlineMarkers'][-1], str(ex))
+        return wflow
+
+    def setPlineMarker(self, wflow, pName=None):
```

Comment:
I have a few comments on this method.
Why do we need to mark (set) which pipeline is executing the workflow? For error reporting?
I also think this is an overcomplication that should be avoided (besides having a fragile implementation).

Reply:
This we discussed in a private call also. As mentioned above I am working on approving it now.

Reply:
Done